### ISSUE TITLE
Rollback does not properly deal with structures like trees

### ISSUE THREAD

**Issue by NotAFile** [piqueserver/piqueserver#214]
If you destroy the bottom of a tree and especially if you build something so it does not fall down before doing that, rollback fails miserably at properly restoring the map

**Comment by NotAFile** [comment]
Turns out this is a BetterSpades bug

**Comment by NotAFile** [comment]
No, it's not just. Betterspades had a bug, but it's broken on vanilla too. Here is the structure to reproduce it:

![image](https://user-images.githubusercontent.com/5447747/34343077-0f359c8c-e9c4-11e7-9247-0f44125f743a.png)

**Comment by 1AmYF** [comment]
This is a client issue.

>Rollback still thinks there is only a single block missing at the bottom of the tree, and only places that

If you reconnect to the server, you will see that the full tree is there. I think the reason is that clients don't distinguish between calls of `remove_point` and `destroy_point`. Only the latter will collapse the whole structure, while `remove_point` is aimed to just remove a block and leave the rest (possibly floating) structure intact. In clients however every block action on something floating (like removal of the red blocks in your screenshot) will result in collapse.
